libraries/AP_Math/crc.cpp

```diff
@@ -539,3 +541,30 @@ uint64_t crc_crc64(const uint32_t *data, uint16_t num_words)
 
     return crc;
 }
+
+// return the parity of byte - "1" if there is an odd number of bits
+// set, "0" if there is an even number of bits set note that
+// __builtin_parity causes hardfaults on Pixracer-periph - and is
+// slower on 1 byte than this:
+uint8_t parity(uint8_t byte)
+{
+    uint8_t p = 0;
+
+    p ^= byte & 0x1;
```

Comment:
why not p ^= byte & 0x2;

Reply:
Tried that's it's bigger and slower as you have to turn the result into a 1 or 0 afterwards